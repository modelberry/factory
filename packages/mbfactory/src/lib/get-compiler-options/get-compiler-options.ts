import ts from 'typescript'

export const getCompilerOptions = () => {
  const configFileName = ts.findConfigFile(
    './',
    ts.sys.fileExists,
    // TODO: Make this a config option
    'tsconfig.json'
  )
  const configFile = ts.readConfigFile(configFileName!, ts.sys.readFile)
  const compilerOptions = ts.parseJsonConfigFileContent(
    configFile.config,
    ts.sys,
    './'
  )
  return compilerOptions
}
