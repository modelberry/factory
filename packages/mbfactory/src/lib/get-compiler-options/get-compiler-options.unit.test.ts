import { getCompilerOptions } from './get-compiler-options'

describe('getCompilerOptions', () => {
  test('should get compiler options', async () => {
    expect(getCompilerOptions().options).toMatchSnapshot()
  })
})
