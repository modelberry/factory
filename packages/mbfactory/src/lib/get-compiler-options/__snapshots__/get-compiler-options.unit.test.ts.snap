// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getCompilerOptions should get compiler options 1`] = `
Object {
  "compileOnSave": false,
  "errors": Array [],
  "fileNames": Array [
    "src/cli.ts",
    "src/plain.ts",
    "src/commands/pull/pull.ts",
    "src/commands/pull/pull.unit.test.ts",
    "src/commands/push/push.ts",
    "src/commands/push/push.unit.test.ts",
    "src/lib/call-handler.ts",
    "src/lib/call-handler.unit.test.ts",
    "src/lib/case-helpers.ts",
    "src/lib/case-helpers.unit.test.ts",
    "src/lib/interface-to-doc-property.ts",
    "src/lib/is-exported-declaration.ts",
    "src/lib/parse-modelberry-interface.ts",
    "src/lib/parse-modelberry-variable.ts",
    "src/lib/get-all-plugin-data/get-all-plugin-data.ts",
    "src/lib/get-all-plugin-data/get-all-plugin-data.unit.test.ts",
    "src/lib/get-compiler-options/get-compiler-options.ts",
    "src/lib/get-compiler-options/get-compiler-options.unit.test.ts",
    "src/research/contentful-management-types.ts",
    "src/research/poc.ts",
  ],
  "options": Object {
    "allowJs": true,
    "allowUmdGlobalAccess": true,
    "configFilePath": undefined,
    "declaration": false,
    "esModuleInterop": true,
    "forceConsistentCasingInFileNames": true,
    "jsx": 4,
    "lib": Array [
      "lib.dom.d.ts",
      "lib.dom.iterable.d.ts",
      "lib.esnext.d.ts",
    ],
    "module": 1,
    "moduleResolution": 2,
    "noImplicitReturns": true,
    "noUnusedLocals": true,
    "removeComments": true,
    "resolveJsonModule": true,
    "skipLibCheck": true,
    "sourceMap": false,
    "strict": true,
    "target": 1,
  },
  "projectReferences": undefined,
  "raw": Object {
    "compileOnSave": false,
    "compilerOptions": Object {
      "declaration": false,
      "sourceMap": false,
    },
    "exclude": Array [
      "**/__fixtures__/**/*",
    ],
    "extends": "../../tsconfig.json",
    "include": Array [
      "src/**/*.ts",
      "src/**/*.tsx",
    ],
  },
  "typeAcquisition": Object {
    "enable": false,
    "exclude": Array [],
    "include": Array [],
  },
  "watchOptions": undefined,
  "wildcardDirectories": Object {
    "src": 1,
  },
}
`;
